src/simd/distances_avx.cc

```diff
@@ -195,5 +151,69 @@ fvec_Linf_avx(const float* x, const float* y, size_t d) {
     return _mm_cvtss_f32(msum2);
 }
 
+// trust the compiler to unroll this properly
+FAISS_PRAGMA_IMPRECISE_FUNCTION_BEGIN
+void
+fvec_madd_avx(size_t n, const float* a, float bf, const float* b, float* c) {
+    FAISS_PRAGMA_IMPRECISE_LOOP
+    for (size_t i = 0; i < n; i++) {
+        c[i] = a[i] + bf * b[i];
+    }
+}
+FAISS_PRAGMA_IMPRECISE_FUNCTION_END
+
+// trust the compiler to unroll this properly
+FAISS_PRAGMA_IMPRECISE_FUNCTION_BEGIN
```

Comment:
Can we make a same macro inside Knowhere can call it Knowhere_xxx_xxx?

Reply:
We can, but I don't see a point of reintroducing a macro, because it is already available in Faiss.

Reply:
Since this will be apply not only FAISS indexes, but also others. So it is a little bit weird to have FAISS word here.